app/lib/utils/decoder_factory.rb

```diff
@@ -0,0 +1,23 @@
+require_relative "secret_decoder"
+require_relative "jwks_decoder"
```

Comment:
I see the problem -- the Rails autoloader requires the path match the module::class name of the object reference, so these would need `module Utils` around the `class SecretDecoder`

Reply:
done.  (thanks for figuring this out @suprjinx )